### PR TITLE
Add useful instances to UnitExp

### PR DESCRIPTION
**Pull request by s9gf4ult**

### CODE REVIEW

**Comment by goldfirere** [comment]
Why just this one datatype? Perhaps we should add `Data` and `Generic` to every declaration?

File: Text/Parse/Units.hs [hunk 1]
```diff
@@ -181,7 +184,7 @@ type UnitTable u = String -> Maybe u
 -- representations.
 data SymbolTable pre u = SymbolTable { prefixTable :: PrefixTable pre
                                      , unitTable   :: UnitTable u
-                                     }
+                                     } deriving (Generic)
```
Comment (s9gf4ult) [inline Text/Parse/Units.hs:187]: `Data` is not deriveable for this type

**Comment by s9gf4ult** [comment]
Added instance for `SymbolTable`, but other types are not exported so I dont see any reason to add instances for.

**Comment by goldfirere** [comment]
Good point. Thanks!

**Pull request merged**